Keycloak.Migrator.DataServices/UserMigrationService.cs

```diff
@@ -0,0 +1,117 @@
+﻿using Keycloak.Migrator.DataServices.Interfaces;
+using Keycloak.Migrator.Models;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Keycloak.Migrator.DataServices
+{
+    public class UserMigrationService : IUserMigrationService
+    {
+        private readonly ILogger<UserMigrationService> _logger;
+        private readonly IUserDataService _userDataService;
+        private readonly IRolesDataService _roleDataService;
+        //private readonly IRolesDataService
+        public UserMigrationService(IUserDataService userDataService,
+            IRolesDataService roleDataService,
+            ILogger<UserMigrationService> logger)
+        {
+            _userDataService = userDataService;
+            _logger = logger;
+            _roleDataService = roleDataService;
+        }
+
+        public async Task<bool> MigrateUsers(ImportMigrationDataJSON jsonData)
+        {
+            var existingUsers = await _userDataService.GetUsers(jsonData.Realm);
+            var roles = await _roleDataService.GetRoles(jsonData.Realm);
+
+            foreach (var user in jsonData.Users)
+            {
+                //If the User doesn't exist in existing Users
+                if(!existingUsers.Any(r => r.Email == user.Email))
+                {
+                    var userSync = new Net.Models.Users.User()
+                    {
+                        UserName = user.User_name,
+                        FirstName = user.First_name,
+                        LastName = user.Last_name,
+                        Email = user.Email,
+                        Enabled = true
+                    };
```

Comment:
We should implement Automapper to perform this mapping.
Add the mapper to the ServiceProviderFactory.